[DSM] PEPPER-883

pegahtah opened the pull request:
## Context

This PR consists of bash scripts for making the necessary changes in the GCP to create pubsub topics and subscriptions, the buckets and their alerts as well as the service account with it's needed IAM policy. 

To deploy the CF, we just need to run `./cf-deploy.sh <env> <study-name>` 


## Checklist

- [x] I have labeled the type of changes involved using the `C-*` labels.
- [x] I have assessed potential risks and labeled using the `R-*` labels.
- [x] I have considered error handling and alerts, and added `L-*` labels as needed.
- [x] I have considered security and privacy, and added `I-*` labels as needed
- [ ] I have analyzed my changes for stability, fault tolerance, graceful degradation, performance bottlenecks and written a brief summary in this PR.
- [ ] If applicable, I have discussed the analytics needs at both a platform and study level with Product and instrumented code accordingly.
- [ ] If applicable, my UI/UX changes have passed muster with Product/Design via an over-the-shoulder review, screenshots, etc.

_If unsure or need help with any of the above items, add the `help wanted` label. For items that starts with `If applicable`, if it is not applicable, check it off and add `n/a` in front._

## FUD Score

_Overall, how are you feeling about these changes?_

- [x] :relaxed: All good, business as usual!
- [x] :sweat_smile: There might be some issues here
- [ ] :scream: I'm sweaty and nervous

## How do we demo these changes?

_How does one observe these changes in a deployed system? Note that **user visible** encompasses many personas--not just patients and study staff, but also ops duty, your fellow devs, compliance, etc._

- [ ] They are user-visible in dev as a regular user journey and require no additional instructions.
- [x] Getting dev into a state where this is user-visible requires some tech fiddling. I have documented these steps in the related ticket.
- [ ] Requires other features before it's human visible. I have documented the blocking issues in jira.
- [ ] I have no idea how to demo this. Please help me!

## Testing

- [ ] I have written automated positive tests
- [ ] I have written automated negative tests
- [x] I have written zero automated tests but have poked around locally to verify proper functionality
- [ ] The jira ticket has acceptance criteria and QA has the needed information to test changes

## Release

- [ ] These changes require no special release procedures--just code!
- [x] Releasing these changes requires special handling and I have documented the special procedures in the release plan document

pegahtah committed:
PEPPER-883 changes for building and deploying

pegahtah committed:
PEPPER-883 changes for building and deploying

pegahtah committed:
PEPPER-883 changes for building and deploying

denniscunningham commented on cloud-functions/cf-dsm-file-scanner/src/main/java/org/broadinstitute/ddp/cf/DSMSomaticFileScanner.java:
```diff
@@ -0,0 +1,334 @@
+import com.google.pubsub.v1.PubsubMessage;
+import lombok.AllArgsConstructor;
+import lombok.Value;
+import lombok.extern.flogger.Flogger;
```
I realize this is a cut and paste, but I don't think we need Flogger. I don't think it is a problem per se, but we don't use it elsewhere, and it does not seem to be buying us anything.

pegahtah replied:
@denniscunningham I agree with you! I tried with Slf4j which is what we use but the logs do not get published correctly. I think it's because Slf4j does not have the same severity levels

denniscunningham commented on cloud-functions/cf-dsm-file-scanner/dsm-gcp-setup/cf-deploy.sh:
```diff
@@ -0,0 +1,342 @@
+    --image="$CONTAINER_FQ_NAME" \
+    --min-instances="1" \
+    --max-instances="3" \
+    --env-vars-file="env.yaml" \
```
Where is `env.yaml`?
I notice it is also missing from cf-file-scanner, but I don't understand how either deployment works without it.

pegahtah committed:
PEPPER-883 changes for building and deploying

pegahtah committed:
PEPPER-883 changes for building and deploying

pegahtah committed:
PEPPER-883
Adding the --no-cpu-throttling option to make sure CPU is always allocated!
Changing the clamav version to 1.1.0
Setting the CLAMD_STARTUP_TIMEOUT to 7200

pegahtah committed:
PEPPER-883
Changing the create_topic_and_sub.sh to not create subscription in case the name is not given. This is used later to create push subscription and notifications for the trigger topic in dsm-setup.sh
Adding the storage.objectAdmin to the service account creation
Adding one more line of logs into DSMSomaticFileScanner

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into PEPPER-883_cloud_function

otchet-broad commented on cloud-functions/cf-dsm-file-scanner/src/main/java/org/broadinstitute/ddp/cf/DSMSomaticFileScanner.java:
```diff
@@ -0,0 +1,334 @@
+
+    private void publishAntivirusResultMessageToDSM(AntivirusMessage antivirusMessage) {
+        if (publisher.isPresent()) {
+            ByteString data = ByteString.copyFrom(antivirusMessage.toString(), StandardCharsets.UTF_8);
```
How about converting to JSON so we send a simple object with just:
{
"bucketId":"\<original bucket\>",
"objectId":"\<original object id\>",
"newBucketId":"\<new bucket\>",
"newObjectId"":"\<new object ID\>",
"scanResult":"\<scan result\>"
}

pegahtah replied:
Implemented

pegahtah committed:
PEPPER-883
Changed the pubsub message to a JSON
Surrounding the move and delete operations with try-catch

pegahtah committed:
Merge remote-tracking branch 'origin/develop' into PEPPER-883_cloud_function

pegahtah committed:
PEPPER-883
Final bucket name updated

otchet-broad commented on cloud-functions/cf-dsm-file-scanner/Dockerfile:
```diff
@@ -0,0 +1,38 @@
+
+ENV GCP_PROJECT="broad-ddp-dev"
+ENV RESULT_TOPIC="dsm-file-antivirus-result"
+ENV STUDY_CLEAN_BUCKET="cmi-lms-files"
```
```suggestion
ENV STUDY_CLEAN_BUCKET="somatic-result-files"
```

pegahtah replied:
fixed! thank you!

pegahtah committed:
PEPPER-883
Removed study for creation, one service for all
adding --region flag for service describe
